cpp/src/arrow/engine/substrait/function_test.cc

```diff
@@ -313,7 +330,18 @@ TEST(FunctionMapping, ValidCases) {
        {"abc", "def"},
        {utf8(), utf8()},
        "abcdef",
-       utf8()}};
+       utf8()},
+      {{kSubstraitLogarithmicFunctionsUri, "ln"}, {"1"}, {int8()}, "0", float64()},
+      {{kSubstraitLogarithmicFunctionsUri, "log10"}, {"1"}, {int8()}, "0", float64()},
+      {{kSubstraitLogarithmicFunctionsUri, "log2"}, {"2"}, {int8()}, "1", float64()},
+      {{kSubstraitLogarithmicFunctionsUri, "log1p"}, {"0"}, {int8()}, "0", float64()},
+      {{kSubstraitLogarithmicFunctionsUri, "logb"},
```

Comment:
I'll note these tests don't have a lot of discriminating power.

Reply:
Agreed. I've changed the `exp(0)`-like cases to be more discriminating.

Also note the comment above:
```
// These are not meant to be an exhaustive test of Substrait
// conformance.  Instead, we should test just enough to ensure
// we are mapping to the correct function
TEST(FunctionMapping, ValidCases) {
```